core/events/events.js

```diff
@@ -14,201 +14,286 @@
  * @const
  */
-Blockly.Events.SELECTED = 'selected';
+const SELECTED = 'selected';
+exports.SELECTED = SELECTED;
 
 /**
  * Name of event that records a click.
  * @const
  */
-Blockly.Events.CLICK = 'click';
+const CLICK = 'click';
+exports.CLICK = CLICK;
 
 /**
  * Name of event that records a marker move.
  * @const
  */
-Blockly.Events.MARKER_MOVE = 'marker_move';
+const MARKER_MOVE = 'marker_move';
+exports.MARKER_MOVE = MARKER_MOVE;
 
 /**
  * Name of event that records a bubble open.
  * @const
  */
-Blockly.Events.BUBBLE_OPEN = 'bubble_open';
+const BUBBLE_OPEN = 'bubble_open';
+exports.BUBBLE_OPEN = BUBBLE_OPEN;
 
 /**
  * Name of event that records a trashcan open.
  * @const
  */
-Blockly.Events.TRASHCAN_OPEN = 'trashcan_open';
+const TRASHCAN_OPEN = 'trashcan_open';
+exports.TRASHCAN_OPEN = TRASHCAN_OPEN;
 
 /**
  * Name of event that records a toolbox item select.
  * @const
  */
-Blockly.Events.TOOLBOX_ITEM_SELECT = 'toolbox_item_select';
+const TOOLBOX_ITEM_SELECT = 'toolbox_item_select';
+exports.TOOLBOX_ITEM_SELECT = TOOLBOX_ITEM_SELECT;
 
 /**
  * Name of event that records a theme change.
  * @const
  */
-Blockly.Events.THEME_CHANGE = 'theme_change';
+const THEME_CHANGE = 'theme_change';
+exports.THEME_CHANGE = THEME_CHANGE;
 
 /**
  * Name of event that records a viewport change.
  * @const
  */
-Blockly.Events.VIEWPORT_CHANGE = 'viewport_change';
+const VIEWPORT_CHANGE = 'viewport_change';
+exports.VIEWPORT_CHANGE = VIEWPORT_CHANGE;
 
 /**
  * Name of event that creates a comment.
  * @const
  */
-Blockly.Events.COMMENT_CREATE = 'comment_create';
+const COMMENT_CREATE = 'comment_create';
+exports.COMMENT_CREATE = COMMENT_CREATE;
 
 /**
  * Name of event that deletes a comment.
  * @const
  */
-Blockly.Events.COMMENT_DELETE = 'comment_delete';
+const COMMENT_DELETE = 'comment_delete';
+exports.COMMENT_DELETE = COMMENT_DELETE;
 
 /**
  * Name of event that changes a comment.
  * @const
  */
-Blockly.Events.COMMENT_CHANGE = 'comment_change';
+const COMMENT_CHANGE = 'comment_change';
+exports.COMMENT_CHANGE = COMMENT_CHANGE;
 
 /**
  * Name of event that moves a comment.
  * @const
  */
-Blockly.Events.COMMENT_MOVE = 'comment_move';
+const COMMENT_MOVE = 'comment_move';
+exports.COMMENT_MOVE = COMMENT_MOVE;
 
 /**
  * Name of event that records a workspace load.
  */
-Blockly.Events.FINISHED_LOADING = 'finished_loading';
+const FINISHED_LOADING = 'finished_loading';
+exports.FINISHED_LOADING = FINISHED_LOADING;
 
 /**
  * Type of events that cause objects to be bumped back into the visible
  * portion of the workspace.
  *
  * Not to be confused with bumping so that disconnected connections do not
  * appear connected.
- * @typedef {!Blockly.Events.BlockCreate|!Blockly.Events.BlockMove|
- * !Blockly.Events.CommentCreate|!Blockly.Events.CommentMove}
+ * @typedef {!BlockCreate|!BlockMove|
+ * !CommentCreate|!CommentMove}
  */
-Blockly.Events.BumpEvent;
+let BumpEvent;
+exports.BumpEvent = BumpEvent;
```

Comment:
This is a typedef, does this need to be a `let` variable?

Reply:
The compiler complains if it's made a `const` because it isn't being assigned a value.